src/Planner/PlannerJoins.cpp

```diff
@@ -805,7 +805,14 @@ static std::shared_ptr<IJoin> tryCreateJoin(JoinAlgorithm algorithm,
         if (table_join->allowParallelHashJoin())
         {
             auto query_context = planner_context->getQueryContext();
-            return std::make_shared<ConcurrentHashJoin>(query_context, table_join, query_context->getSettings().max_threads, right_table_expression_header);
+            const auto & settings = query_context->getSettingsRef();
+            StatsCollectingParams params{
+                calculateCacheKey(table_join, right_table_expression),
+                settings.collect_hash_table_stats_during_aggregation,
+                query_context->getServerSettings().max_entries_for_hash_table_stats,
+                settings.max_size_to_preallocate_for_aggregation};
```

Comment:
Let's use separate settings for join. To see `max_size_to_preallocate_for_aggregation` here is confusing